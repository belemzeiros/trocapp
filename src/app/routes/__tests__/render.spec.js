import renderRoutes, { layoutHoc } from '../render';

const routesMock = [
  {
    layout: jest.fn(),
    page: jest.fn(),
    path: '/',
    exact: true,
  },
  {
    page: jest.fn(),
    path: '/without-layout',
    exact: true,
  },
];

describe('Render Routes', () => {
  it('Should rendered routes successfully', () => {
    const wrapper = renderRoutes(routesMock);
    expect(wrapper).toMatchSnapshot();
  });

  it('When routes is null', () => {
    const wrapper = renderRoutes(null);
    expect(wrapper).toBeNull();
  });

  it('Should render defined hoc', () => {
    layoutHoc(routesMock[0], {})({});
    layoutHoc(routesMock[1], {})({});
  });
});
