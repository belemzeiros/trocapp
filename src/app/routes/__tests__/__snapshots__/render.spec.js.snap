// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Routes Should rendered routes successfully 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/without-layout"
    render={[Function]}
  />
</Switch>
`;
